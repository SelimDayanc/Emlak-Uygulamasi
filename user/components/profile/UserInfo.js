/**
 @flow
 */
import React, {Component, PropTypes} from 'react';
import {ScrollView, StyleSheet, Text, View} from 'react-native';
import colors from '../../../common/colors';
import Separator from './../../../components/Separator';

export default class UserInfo extends Component {
  static propTypes = {
    user: PropTypes.object.isRequired,
  };

  render() {
    const {user} = this.props;

    return (
      <ScrollView style={styles.container}>
        <View style={styles.rowContainer}>

          <Text style={styles.description}>
            {user.company.description}
          </Text>
          <Separator style={{marginVertical: 20}} />
        </View>
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 10,
    backgroundColor: 'white',
    paddingTop: 20,
  },
  description: {
    fontWeight: '100',
    color: colors.darkGrey,
    fontSize: 15,
  },
  content: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  name: {
    fontSize: 16,
    fontWeight: '100',
    color: colors.darkGrey,
  },
  label: {
    paddingBottom: 5,
    color: colors.smokeGreyLight,
  },
  icon: {
    width: 20,
    height: 20,
  },
});
