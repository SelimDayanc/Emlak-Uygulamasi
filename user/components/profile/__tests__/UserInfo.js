import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import UserInfo from './../UserInfo';

test('renders User Info', () => {
  const tree = renderer
    .create(
      <UserInfo
        user={{
          name: 'name',
          email: 'z4ls@live.com',
          isCompany: true,
          company: {description: 'Company Description'},
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
