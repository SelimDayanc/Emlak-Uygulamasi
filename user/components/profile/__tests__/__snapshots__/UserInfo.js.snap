// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders User Info 1`] = `
<RCTScrollView
  alwaysBounceHorizontal={undefined}
  alwaysBounceVertical={true}
  onContentSizeChange={null}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={undefined}
  onResponderTerminationRequest={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onScrollShouldSetResponder={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  sendMomentumEvents={false}
  style={
    Array [
      Object {
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "scroll",
      },
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "padding": 10,
        "paddingTop": 20,
      },
    ]
  }
>
  <RCTScrollContentView
    collapsable={false}
    removeClippedSubviews={undefined}
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#484848",
            "fontSize": 15,
            "fontWeight": "100",
          }
        }
      >
        Company Description
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#e3e7ed",
              "height": 0.5,
            },
            Object {
              "marginVertical": 20,
            },
          ]
        }
      />
    </View>
  </RCTScrollContentView>
</RCTScrollView>
`;
